test/build-with-timeout/build-with-low-timeout.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: build.knative.dev/v1alpha1
+kind: Build
+metadata:
+  name: test-with-low-timeout
+  labels:
+    expect: failed
+timeout: 50s
+spec:
+  steps:
+  - name: step1
+    image: ubuntu
+    command: ["/bin/bash"]
+    args: ["-c", "sleep 2000"]
```

Comment:
Oh okay, so this test indicates that `syncHandler` must be called periodically, or at least not only when steps start/finish. Do we have any idea when or how often exactly `syncHandler` is called?

Reply:
30 seconds is the sync interval